File: assignments_nilesh_sir/java_assignments/Assignment_Bank/Banking/PersonalLoan.java
package Banking;

public class PersonalLoan extends Loan implements Taxable {
    public PersonalLoan() {
    }

    public double getRate() {
        return getPrinciple() >= 500000 ? 0.15 : 0.16;
    }

    public double getTax() {
        return getEMI() > 1500 ? getEMI() * 0.05 : 0.0;
    }
}

File: Dotnet/04-05-24/Assignment_3/BankLib/PersonalLoan.cs
namespace BankLib
{
    sealed class PersonalLoan : Loan, ITaxable
    {
        public override double GetRate()
        {
            double lowerlimit = 300000;
            double upperlimit = 2500000;
            double rate = Principle >= 500000 ? 0.15 : 0.16;
            if (Principle < lowerlimit)
            {
                return rate += 0.01;
            }
            else if (Principle > upperlimit)
            {
                return rate += 0.02;
            }
            else
            {
                return rate;
            }
        }

        public double GetTax()
        {
            return GetEmi() * 0.10;
        }
    }
}